File: BackendVerticalSlice/LabResource.VerticalSliceAPI/Common/Entities/LabAsset.cs
using LabResource.VerticalApi.Common.Enums;

namespace LabResource.VerticalApi.Common.Entities;

public class LabAsset
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? SerialNumber { get; set; }

    public AssetStatus Status { get; set; } = AssetStatus.Available;

    public bool IsActive { get; set; } = true; 

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;


    public ICollection<BorrowingRecord> BorrowingRecords { get; set; } = new List<BorrowingRecord>();
}

File: FrontendAngular/src/app/core/models/lab-asset.ts
import { AssetStatus } from './enums';

export interface LabAsset {
  id: string;
  name: string;
  serialNumber?: string | null;
  location?: string | null;
  status: AssetStatus;
  isActive: boolean;
  assignedTeacherId?: string | null;
  assignedTeacherName?: string | null;
  currentBorrowerName?: string | null;
  currentBorrowDate?: string | null;
}

export interface LabAssetResponse extends LabAsset {}

export interface CreateLabAssetRequest {
  name: string;
  serialNumber?: string | null;
  location?: string | null;
  assignedTeacherId?: string | null;
}

export interface UpdateLabAssetRequest {
  name: string;
  serialNumber?: string | null;
  location?: string | null;
  assignedTeacherId?: string | null;
  isDefective: boolean;
}